### DISCUSSION TITLE
Add Battery stock management

### DISCUSSION THREAD

**Discussion by McFlyPartages** [andrew-codechimp/HA-Battery-Notes#650] [Ideas]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/andrew-codechimp/ha-battery-notes/issues?q=is%3Aopen+is%3Aissue+label%3Aenhancement).

### Is your feature request related to a problem? Please describe.

No more fresh batteries

### Describe the solution you'd like

An interesting function would be the management of the stock of the different stacks of the network with a decrease each time we press PRESS of an entity when we change the stack.

This could be used to launch a callback when there are less than X stacks left.

Thanks for the current integration, which is already a fabulous piece of work.

### Describe alternatives you've considered

None

### Additional context

None

**Comment by andrew-codechimp** [comment]
This is something I've thought about a bit and has been discussed in the past.
This would be a long term improvement, I'll move it into the discussions area so not forgotten and hopefully we'll get some more feedback on what might be wanted on this.

**Reply by McFlyPartages** [reply]
Great, thank you very much

**Comment by nickwest** [comment]
I had a similar thought to this when I stumbled across this integration. The proposed battery stock management idea is nice, I think that would be useful. My initial thought was a custom card to display the types number of deployed batteries in a single view and number of batteries consumed, based on replacement count.

Adding a way to track battery cost with the proposed battery stock management would be great too. I imagine attaching a cost field to a "restocking" function would be a practical way to go about that given costs will vary wildly.

**Comment by andrew-codechimp** [comment]
I think it would need a whole new dashboard rather than a custom card.  Something I'm looking into but it's a very different skillset to create those compared with integrations.
I'm thinking something a bit like HACS experimental mode, a device list on steroids with action menu's and maybe mixing in a bit of something similar to the energy dashboard with multiple cards.
This isn't forgotten and I think the main integration will be settling down a bit so I'll get chance to explore this more.
